### PR TITLE
damoreno robot-name solution

### PR DESCRIPTION
**Pull request by davidmogar**

### CODE REVIEW

File: robot-name/damoreno/robot_name.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import "fmt"
+
+type Robot struct {
+	Id string
+}
+
+var availableRobotNames = func() map[string]struct{} {
+	names := make(map[string]struct{})
+
+	for firstLetter := 'A'; firstLetter <= 'Z'; firstLetter++ {
```
Comment (enarha) [inline robot-name/damoreno/robot_name.go:12]: As I commented on another MR, the exercise says that the names must be random "The names must be random: they should not follow a predictable sequence", while what you do here is pre-generating the names in certain order and you also serve them in that order.

File: robot-name/damoreno/robot_name.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		for secondLetter := 'A'; secondLetter <= 'Z'; secondLetter++ {
+			for number := 0; number < 1000; number++ {
+				robotName := fmt.Sprintf("%c%c%03d", firstLetter, secondLetter, number)
+				names[robotName] = struct{}{}
```
Comment (enarha) [inline robot-name/damoreno/robot_name.go:16]: What does `struct{}{}` do, I mean what's the meaning in your code?

Comment (fnasser) [reply robot-name/damoreno/robot_name.go:16]: “names” is a set (which is a hack in Go implemented with a map).  The code is only interested in storing the value with robotName (as a key), and does not care about any value.  So a type with no fields, so no value, is used as the map value as it takes no space.

Comment (enarha) [reply robot-name/damoreno/robot_name.go:16]: Thanks for the insight @fnasser

**Pull request merged**